controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -814,36 +813,33 @@ func (r *StorageClassClaimReconciler) reconcileCephFilesystemSubVolumeGroup() er
 		},
 	}
 	if err := r.get(&cephFilesystem); err != nil {
-		return fmt.Errorf("error fetching CephFilesystemList. %+v", err)
+		return fmt.Errorf("error fetching CephFilesystem. %+v", err)
 	}
 
 	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephFilesystemSubVolumeGroup, func() error {
 		if err := r.own(r.cephFilesystemSubVolumeGroup); err != nil {
 			return err
 		}
 		deviceClass := r.storageProfile.DeviceClass
-		cephFileSystemName := ""
-		dataPoolName := ""
+		var dataPool rookCephv1.NamedPoolSpec
 		for i := range cephFilesystem.Spec.DataPools {
 			if cephFilesystem.Spec.DataPools[i].DeviceClass == deviceClass {
-				cephFileSystemName = cephFilesystem.Name
-				dataPoolName = cephFilesystem.Spec.DataPools[i].Name
+				dataPool = cephFilesystem.Spec.DataPools[i]
```

Comment:
This will create an in-memory copy of the data pool, it will be preferable to take a pointer.
```suggestion 
				dataPool = &cephFilesystem.Spec.DataPools[i]
```